### PR TITLE
fix build with with GO111MODULE=off

### PR DESCRIPTION
**Pull request by lsm5**
Fix build with GO111MODULE=off
    
Distro builds on Fedora and Kubic projects use GO111MODULE=off
by default which are currently failing. This commit fixes it and
going forward, podman CI will also indicate failures in rpm builds.

conmon build has also been removed from podman.spec.in because the COPR
for which it was provided has been discontinued.

Fixes: #10009

Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

@mheon @vrothberg @rhatdan @baude @cevich PTAL

### CODE REVIEW

**Comment by lsm5** [comment]
let me add a CI job for GO111MODULE=off

**Comment by mheon** [comment]
This seems OK, LGTM

**Review by vrothberg — COMMENTED** [review]
LGTM

**Comment by lsm5** [comment]
Let's hold off on lgtm until we have a green CI.

**Comment by cevich** [comment]
@lsm5 I'm curious why this is needed.  My understanding is `GO111MODULE=off` is only necessary when using really, really, old versions of golang, no?

---

CI test-job wise, this sounds like something ripe for inclusion in `alt_build_task`.  This is a kind of catch-all for "Confirm building works in *" scenarios.  To add something new, just toss in a new matrix item to that task with a distinct `$ALT_NAME` value.  Then pick that up with a [new entry in the runner script case block.](https://github.com/containers/podman/blob/b6b0b6e8bd0f8c75fe9411f722a43f9305c6137a/contrib/cirrus/runner.sh#L208-L250)

**Comment by lsm5** [comment]
> @lsm5 I'm curious why this is needed. My understanding is `GO111MODULE=off` is only necessary when using really, really, old versions of golang, no?

hmm, I never bothered to check honestly but I could dig in if necessary on that. FWIW, the default fedora `%gobuild` definition is:

```
$ rpm --eval %gobuild

  # https://bugzilla.redhat.com/show_bug.cgi?id=995136#c12
     GO111MODULE=off \
  go build -buildmode pie -compiler gc -tags="rpm_crashtraceback ${BUILDTAGS:-}" -ldflags "${LDFLAGS:-} -B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \n') -extldflags '-Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld  '" -a -v -x ;
```

so I'm guessing there hasn't been any change on newer versions.

> CI test-job wise, this sounds like something ripe for inclusion in `alt_build_task`. This is a kind of catch-all for "Confirm building works in *" scenarios. To add something new, just toss in a new matrix item to that task with a distinct `$ALT_NAME` value. Then pick that up with a [new entry in the runner script case block.](https://github.com/containers/podman/blob/b6b0b6e8bd0f8c75fe9411f722a43f9305c6137a/contrib/cirrus/runner.sh#L208-L250)

So, given that the spec file is only used for checking if rpm builds fine, I changed the spec file directly without a specific CI job. Things should hopefully be changing very soon when we have the packit integration in place.

**Comment by lsm5** [comment]
@cevich oh also, if we don't have GO111MODULE=off it ends up fetching all the vendored dependencies which is a no-go for non-networked koji buiilds.

**Comment by cevich** [comment]
> https://bugzilla.redhat.com/show_bug.cgi?id=995136

That BZ is from 2016.

> if we don't have GO111MODULE=off it ends up fetching all the vendored dependencies which is a no-go for non-networked koji buiilds.

Oh...right, yeah, golang really wants to be connected to the internet :disappointed:

**Comment by cevich** [comment]
> Things should hopefully be changing very soon when we have the packit integration in place.

Oh okay, yeah, feel free to replace that RPM-build job then.  My understand is it's mainly there for package release benefit - noticing problems early.  So if there's a better way to do that, go for it.

**Comment by lsm5** [comment]
`Test build RPM` is successful with current commit, so I'm gonna push another version with `[NO TESTS NEEDED]` to handle `Validate fedora-34beta Build`. Going forward, I think there should be exceptions for cases where no tests are added/modified but tests are still required to pass.

**Comment by lsm5** [comment]
ready for lgtm

File: pkg/domain/entities/types/auth.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package types // import "github.com/docker/docker/api/types"
```
Comment (rhatdan) [inline pkg/domain/entities/types/auth.go:1]: Why did you change these?

Comment (lsm5) [reply pkg/domain/entities/types/auth.go:1]: that's the main thing that breaks build with `GO111MODULE=off`. Please let me know if there's a better option to handle that.

**Comment by mheon** [comment]
I think the comment from @rhatdan is addressed
/lgtm

**Comment by lsm5** [comment]
we also need a `/hold cancel` i guess.

**Comment by rhatdan** [comment]
I thought those were there to indicate where we copied the code from .  Perhaps change the import to copied from?

**Comment by lsm5** [comment]
> I thought those were there to indicate where we copied the code from . Perhaps change the import to copied from?

@rhatdan I can give that a try.

**Comment by lsm5** [comment]
@rhatdan added a comment on top of the package line. I wish golang didn't act funny with comments.

**Comment by lsm5** [comment]
anybody know how to rerun the rdoproejct.org/github-check job? It doesn't fail locally for me on centos8

**Comment by edsantiago** [comment]
rdoproject always fails. We don't care: it doesn't block merging.

**Comment by lsm5** [comment]
> rdoproject always fails. We don't care: it doesn't block merging.

thanks @edsantiago . If someone could  please `/hold cancel`, this should be good for merge.
